src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -130,24 +131,48 @@ private static void ModifyUnmanagedDllResolver(Action<AssemblyLoadContext> resol
 
         private static IntPtr HostFxrResolver(Assembly assembly, string libraryName)
         {
+            Console.Error.WriteLine($"Try to load native library {libraryName}");
+
             var hostFxrLibName = "libhostfxr";
             var libExtension = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "dylib" : "so";
 
-            if (!hostFxrLibName.Equals(libraryName))
+            // the DllImport hardcoded the name as hostfxr.
+            if (!hostFxrLibName.Equals(libraryName, StringComparison.Ordinal) && !libraryName.Equals("hostfxr", StringComparison.Ordinal))
+            {
+                return IntPtr.Zero;
+            }
+
+            string? dotnetFolderPath = Path.GetDirectoryName(DotnetPath.Value);
```

Comment:
this check is present in https://github.com/lifengl/MSBuildLocator/blob/937a06d622caf3a0fd0326479375700ea1e93932/src/MSBuildLocator/DotNetSdkLocationHelper.cs#L234C18-L234C18

Reply:
actually, I misunderstood the issue, and DotNetPath is the folder path, so I should revert that change. The problem here is DOTNET_HOST_PATH should point to the file, not folder, which lead to some confusions. After fixing that, it seems to work. I also removed excessive logging code added earlier.